vms/platformvm/state/state.go

```diff
@@ -834,9 +904,20 @@ func (s *state) GetSubnetOwner(subnetID ids.ID) (fx.Owner, error) {
 		return nil, fmt.Errorf("%q %w", subnetID, errIsNotSubnet)
 	}
 
+	s.SetSubnetOwner(subnetID, owner)
 	return subnet.Owner, nil
 }
 
+func (s *state) SetSubnetOwner(subnetID ids.ID, owner fx.Owner) {
+	if s.subnetOwners == nil {
+		s.subnetOwners = map[ids.ID]fx.Owner{
+			subnetID: owner,
+		}
+	} else {
+		s.subnetOwners[subnetID] = owner
+	}
```

Comment:
We never `nil` out `subnetOwners` - so we should probably just initialize it once and then remove this check